### DISCUSSION TITLE
citation key title with a hyphen

### DISCUSSION THREAD

**Discussion by Y-Akihiro** [retorquere/zotero-better-bibtex#2643] [General]
I'm using zotero (ver. 6.0.27) on a Windows 10 PC.

Before installing `Better BibTeX`, the citation key was like

>author_title_year

where the `title` included a hyphen.

For example, if the first word in the title is `hyphenated-title`, the citation key was like:

> author_hyphenated-title_year

However, after installing Better BibTeX and setting a citation key formula to be `[auth:lower]_[veryshorttitle:lower]_[year]`, I ended up getting the citation key without a hyphen:

>author_hyphenatedtitle_year

How can I keep a hyphen in the title? Is there an easy fix by changing `veryshorttitle` or `shorttitle`?

**Comment by retorquere** [comment]
If you want the zotero citekeys, why not just use `zotero` in the formula field?

**Reply by Y-Akihiro** [reply]
I have two articles that were published in the same year by the same author and share the same initial word in their titles.
To differentiate them, I wanted to customize the cite keys with Better BibTeX.

**Reply by retorquere** [reply]
Differentiate them how? You can use anything else you want in combination with `zotero`, that doesn't have to be used on its own.

**Reply by retorquere** [reply]
`﻿﻿Title.skipwords.select(1, 1)` should do what you want.

**Comment by DrEntropy** [comment]
One reason you might want to use BBT and keep zotero cite keys is if you want to take advantage of *other* features of BBT, for example excluding fields form export, BUT you already have a huge mulit document report filled with citation keys with the zotero format.   Is there just no way to get BBT to keep the hyphens?

**Reply by retorquere** [reply]
BBT does not remove hyphens unless you tell it to, but the citation keys generated by Zotero's bibtex exporter don't have hyphens (unless you have hyphens in your title), they have underscores. If you want the default Zotero-generated citation keys use `zotero` as your citekey formula. In the case posted about, the hyphen is not removed.

**Reply by DrEntropy** [reply]
OY!  I *totally* misunderstood!   I didn't realize it was as easy as using 'zotero'  as the formula.   Thanks for the fast response and your patience.   (I thought you were saying, in effect, just use zotero, without BBT. )    Color me embarrassed.

**Reply by retorquere** [reply]
yeah I didn't express myself very clearly there.
